scheduler/src/cook/mesos/api.clj

```diff
@@ -844,11 +844,23 @@
             application (assoc :application (util/remove-datomic-namespacing application))
             expected-runtime (assoc :expected-runtime expected-runtime))))
 
-(defn fetch-group-job-details
+(defn fetch-group-jobs-by-state
   [db guuid]
   (let [group (d/entity db [:group/uuid guuid])
-        jobs (:group/job group)
-        jobs-by-state (group-by :job/state jobs)]
+        jobs (:group/job group)]
+    (group-by :job/state jobs)))
+
+(defn fetch-group-live-jobs
+  [db guuid]
+  (->> guuid
+       (fetch-group-jobs-by-state db)
+       ((juxt :job.state/running :job.state/waiting))
+       (apply concat)
+       (vec)))
```

Comment:
Could we just `filter` the jobs by state rather than `group-by`, `juxt`, `concat`?

Reply:
fixed